wait_until_services_up.py

```diff
@@ -0,0 +1,41 @@
+import requests
+from requests import RequestException
+from retrying import retry
+
+from config import Config
+
+
+class HealthCheckException(Exception):
+    def __init__(self, port):
+        self.port = port
+
+    def __str__(self) -> str:
+        return f'Healthcheck fails on port {self.port}'
+
+
+def retry_if_http_error(exception):
+    print(f'error has occurred: {str(exception)}')
+    return isinstance(exception, RequestException) or isinstance(exception, HealthCheckException)
+
+
+@retry(retry_on_exception=retry_if_http_error, wait_fixed=10000, stop_max_delay=600000, wrap_exception=True)
+def check_services():
+    for port in [Config.ACTION_SERVICE_PORT, Config.BACKSTAGE_SERVICE_PORT,
```

Comment:
Using the ports makes sense locally but will this be suitable for testing deployment? (Might make more sense to list the full service config variable for each instead)

Reply:
You could do:

```Python
def check_status(port):
    try:
        resp = requests.get(f'http://localhost:{port}/info')
        resp.raise_for_status()
    except Exception:
        raise HealthCheckException(port)

[check_status(k) for k, v in dict(vars(Config)).items() if '_PORT' in k]
```

Reply:
Ah, nice. Can't do this now as it points to the whole url unless I change the variable names but this is pretty nice